contracts/BananaAllocator.sol

```diff
@@ -0,0 +1,235 @@
+            syncAllocations();
+        }
+    }
+
+    /// @notice A function to withdraw from an allocation to another address
+    /// @param aid number referencing the applicable allocation
+    /// @param amount the amount of tokens to be withdrawn
+    /// @param to the address to withdraw tokens to
+    function withdrawAllocationTo(
+        uint256 aid, 
+        uint256 amount, 
+        address to
+    ) external onlyAllocationAdmin(aid, msg.sender) {
+        if(amount > 0 && amount <= allocationInfo[aid].tokensAvailable) {
+            allocationInfo[aid].tokensAvailable = allocationInfo[aid].tokensAvailable.sub(amount);
+            allocationToken.transfer(to, amount);
+
+            uint256 allocationTokenUsdValue = priceGetter.getTokenUsdPrice(address(allocationToken));
+            uint256 usdAmount = allocationTokenUsdValue.mul(amount).div(1e18);
+
+            emit AllocationWithdrawn(aid, amount, usdAmount, msg.sender);
+        }
+
+        if (lastSyncTimestamp + syncDelay < block.timestamp) {
+            syncAllocations();
+        }
+    }
+
+    /// @notice An internal function to sync all the allocation amounts
+    function syncAllocations() internal {
+        uint256 length = allocationInfo.length;
+        uint256 totalTokensOwed = 0;
+        uint256 tokensInContract = allocationToken.balanceOf(address(this));
+
+        for (uint256 aid = 0; aid < length; aid++) {
+            totalTokensOwed = totalTokensOwed.add(allocationInfo[aid].tokensAvailable);
+        }
+
+        uint256 tokensToAllocate = tokensInContract.sub(totalTokensOwed);
+        
+        for (uint256 i = 0; i < length; i++) {
+            allocationInfo[i].tokensAvailable += allocationInfo[i].allocationAmount.mul(tokensToAllocate).div(totalAllocations);
+        }
+
+        lastSyncTimestamp = block.timestamp;
+        
+        emit AllocationsSynced();
+    }
+
+    /// @notice A function to adjust the sync delay
+    /// @param _syncDelay number to adjust the sync delay to
+    ///     in milliseconds
+    function setSyncDelay(uint256 _syncDelay) external onlyOwner {
+        syncDelay = _syncDelay;
+
+        emit SyncDelaySet(_syncDelay);
+    }
+
+    /// @notice A function to add a new allocation
+    /// @param allocationName string referencing the name of an allocation
+    /// @param allocationAmount amount for allocation to recieve
+    /// @param newAllocationAdmins array of addresses to be admins in this specific allocation
+    function addAllocation(
+        string memory allocationName, 
+        uint256 allocationAmount, 
+        address[] memory newAllocationAdmins
+    ) external onlyOwner {
+        totalAllocations += allocationAmount;
+
+        allocationInfo.push(AllocationInfo({
+            allocationName: allocationName,
+            allocationAmount: allocationAmount, 
+            tokensAvailable: 0
+        }));
+
+        uint256 _aid = allocationInfo.length - 1;
```

Comment:
This can be rewritten as the following. `.push()` returns the new length of the array, so we can subtract 1 at the end and find the current `aid`

```
uint256 _aid = allocationInfo.push(AllocationInfo({
            allocationName: allocationName,
            allocationAmount: allocationAmount, 
            tokensAvailable: 0
        })) - 1;
```